### PR TITLE
Variable name i18n

### PR DESCRIPTION
**Pull request by AnmAtAnm**
Allow string table / Blockly.Msg references in the JSON "variable" attribute.

### CODE REVIEW

**Comment by AnmAtAnm** [comment]
Updated per offline conversation. PTAL @rachel-fenichel

File: core/block.js [hunk 1]
```diff
@@ -1215,6 +1211,70 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
 };
 
 /**
+ * Helper function to construct a FieldImage from a JSON arg object,
+ * dereferncing any string table references.
+ * @param {!Object} options A set of options (src, width, height, and alt).
+ * @returns {!Blockly.FieldImage} The new image.
```
Comment (rachel-fenichel) [inline core/block.js]: @private

File: core/block.js [hunk 2]
```diff
@@ -1215,6 +1211,70 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
 };
 
 /**
+ * Helper function to construct a FieldImage from a JSON arg object,
+ * dereferncing any string table references.
```
Comment (rachel-fenichel) [inline core/block.js]: misspelled dereferencing

File: core/block.js [hunk 3]
```diff
@@ -1215,6 +1211,70 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
...
+ * @param {!Object} options A set of options (src, width, height, and alt).
```
Comment (rachel-fenichel) [inline core/block.js]: A JSON object with options (src, width...

File: core/block.js (hunk 2 above)
Comment (rachel-fenichel) [inline core/block.js]: Same below.

File: core/block.js (hunk 1 above)
Comment (rachel-fenichel) [inline core/block.js]: Same below.

File: core/block.js (hunk 3 above)
Comment (rachel-fenichel) [inline core/block.js]: Same below.

File: core/block.js [hunk 4]
```diff
@@ -1215,6 +1211,70 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
...
+
+/**
+ * Helper function to construct a FieldVariable from a JSON arg object,
+ * dereferncing any string table references.
+ * @param {!Object} options A set of options (variable).
+ * @returns {!Blockly.FieldImage} The new image.
+ */
+Blockly.Block.newFieldVariableFromJson_ = function(options) {
+  var varname = options['variable'];
+  if (goog.isString(varname)) {
```
Comment (rachel-fenichel) [inline core/block.js]: I think it'll always be a string or nothing, so you can just pass directly to Blockly.utils.replaceMessageReferences.

File: core/block.js [hunk 5]
```diff
@@ -1215,6 +1211,70 @@ Blockly.Block.prototype.interpolate_ = function(message, args, lastDummyAlign) {
...
+    varname = Blockly.utils.replaceMessageReferences(varname);
+  }
+  return new Blockly.FieldVariable(varname);;
+};
+
+
```
Comment (rachel-fenichel) [inline core/block.js]: Nit: only one newline.

File: core/block.js (hunk 2 above)
Comment (AnmAtAnm) [inline core/block.js]: Fixed.

File: core/block.js (hunk 3 above)
Comment (AnmAtAnm) [inline core/block.js]: Done.

File: core/block.js (hunk 1 above)
Comment (AnmAtAnm) [inline core/block.js]: done.

File: core/block.js (hunk 4 above)
Comment (AnmAtAnm) [inline core/block.js]: Now handles undefined, null, and really any other type.

**Pull request merged**